OpenRA.Mods.Common/Traits/DamagedByTerrain.cs

```diff
@@ -77,7 +74,21 @@ void INotifyAddedToWorld.AddedToWorld(Actor self)
 			// Cast to long to avoid overflow when multiplying by the health
 			damageThreshold = (int)((Info.DamageThreshold * (long)health.MaxHP + (100 - Info.DamageThreshold) * safeTiles * (long)health.MaxHP / totalTiles) / 100);
 
-			// Actors start with maximum damage applied
+			// TraitEnabled will deal damage if the trait starts disabled and gets enabled later on
```

Comment:
I see two issues here:
* The `StartOnThreshold` description specifically mentions on creation - can we please update this to match the new behaviour?
* AddedToWorld isn't just on creation - capturing a building or units leaving a transport will also trigger the damage effect. Can we add a flag to make sure the damage is only applied the first time?

Allowing the damage to be applied on trait enable instead of just creation confuses matters regarding that second point. Do we really want the actor to be re-damaged if the trait is disabled then restored after creation? Do we need the `AddedToWorld` checks at all, or can this all be done via `TraitEnabled`? Does it make sense for this to stay inside `DamagedByTerrain` at all now? (since you may want to toggle the trait to reset the damage level, independently from the continuous damage effect).

Reply:
I wonder whether it would make more sense to remove `StartOnThreshold` completely, and instead implement the initial damage via a `HealthInit` on a d2k specific subclass of `PlaceBuilding` - this would fit with the d2k's custom PBOG overrides.

Reply:
> Do we need the `AddedToWorld` checks at all, or can this all be done via `TraitEnabled`?

No, because the latter runs before the former for normally created actors. I haven't tested if we occupy the correct cells before being added to the world, but relying on that would be weird regardless.

I forgot about the capturing case, but thought it would make sense for cargo. Anyway, you're right that it makes more sense to just remove `StartOnThreshold`.